internal/daemon/vendor-specific-plugins/marvell/main.go

```diff
@@ -255,11 +360,45 @@ func (vsp *mrvlVspServer) GetDevices(ctx context.Context, in *pb.Empty) (*pb.Dev
 	}, nil
 }
 
+// SetNumVfs function to set the number of VFs with the given context and VfCount
+func (vsp *mrvlVspServer) SetNumVfs(ctx context.Context, in *pb.VfCount) (*pb.VfCount, error) {
+	klog.Infof("Received SetNumVfs() request: VfCnt: %v", in.VfCnt)
+	if vsp.isDPUMode {
+		return nil, errors.New("SetNumVfs is not supported in DPU Mode")
+	}
+	pciAddress, err := mrvlutils.GetPCIByDeviceID(HostDeviceID)
+	if pciAddress == "" || err != nil {
+		return nil, errors.New("PCI Address not found")
+	}
+	vfcnt := in.VfCnt
+	if vfcnt < 0 {
+		return nil, errors.New("invalid VF Count")
+	}
+
+	// reset sriov_numvfs to 0 before setting to a number
+	resetCmd := exec.Command("sh", "-c", fmt.Sprintf("echo 0 > /sys/bus/pci/devices/%s/sriov_numvfs", pciAddress))
+	_, err = resetCmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("failed to reset sriov_numvfs to 0: %v", err)
+	}
```

Comment:
This looks better to me. Obviously, it needs to be tested whether it works well.

Resetting to zero is quite disruptive. Maybe the code should first read the file, and do nothing if it already contains the desired number.